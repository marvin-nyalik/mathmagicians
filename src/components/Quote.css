.quote {
  font-size: 15px;
  font-weight: 400;
  width: 50vw;
  height: auto;
  padding: 1rem;
  margin: 1rem auto;
  border: 1px solid grey;
  border-radius: 5px;
}