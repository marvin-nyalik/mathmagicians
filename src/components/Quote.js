import React, { useEffect, useState } from 'react';
import './Quote.css';

const Quotes = () => {
  const [data, setData] = useState('');
  const [error, setError] = useState('');

  useEffect(() => {
    const fetchQuote = async () => {
      try {
        const quote = await fetch('https://api.api-ninjas.com/v1/quotes?',
          {
            method: 'GET',
            headers: {
              'X-Api-Key': '8m1bnnM+p0GqxgD5kNoIgA==4OYoBSs5rh1rrEan',
            },
          });
        const quoteObj = await quote.json();
        setData(quoteObj[0].quote);
      } catch (error) {
        setError(error.message);
      }
    };
    fetchQuote();
  }, []);

  return (
    <div>
      { data ? <p className="quote">{data}</p> : <p>Loading</p> }
      { error ? <p>{error}</p> : '' }
    </div>
  );
};

export default Quotes;
