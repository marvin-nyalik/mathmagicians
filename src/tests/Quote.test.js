import React from 'react';
import { render } from '@testing-library/react';
import Quote from '../components/Quote';

test('renders Quotes component correctly', () => {
  const { container } = render(<Quote />);
  expect(container.firstChild).toMatchSnapshot();
});
