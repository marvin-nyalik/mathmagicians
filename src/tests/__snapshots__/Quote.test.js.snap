// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Quotes component correctly 1`] = `
<div
  class="container"
>
  <p
    class="loading"
  >
    Loading...
  </p>
</div>
`;
